core/src/main/java/org/elasticsearch/index/codec/PerFieldMappingPostingFormatCodec.java

```diff
@@ -56,7 +57,11 @@ public PerFieldMappingPostingFormatCodec(Lucene50StoredFieldsFormat.Mode compres
     public PostingsFormat getPostingsFormatForField(String field) {
         final MappedFieldType indexName = mapperService.indexName(field);
         if (indexName == null) {
-            logger.warn("no index mapper found for field: [{}] returning default postings format", field);
+            // the query terms are an percolator implementation detail and are added as Lucene indexed fields which are hidden
```

Comment:
@martijnvg correct me if I'm wrong, but this will also remove the need for a custom listener method to parse when reading from the translog, right?